source/extensions/filters/listener/connect_handler/connect_handler.h

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <http_parser.h>
+
+#include "envoy/event/file_event.h"
+#include "envoy/event/timer.h"
+#include "envoy/network/filter.h"
+#include "envoy/stats/scope.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/common/logger.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace ConnectHandler {
+
+/**
+ * All stats for the connect handler. @see stats_macros.h
+ */
+#define ALL_CONNECT_HANDLER_STATS(COUNTER)                                                         \
+  COUNTER(read_error)                                                                              \
+  COUNTER(write_error)                                                                             \
+  COUNTER(connect_not_found)                                                                       \
+  COUNTER(connect_found)
+
+/**
+ * Definition of all stats for the connect handler. @see stats_macros.h
+ */
+struct ConnectHandlerStats {
+  ALL_CONNECT_HANDLER_STATS(GENERATE_COUNTER_STRUCT)
+};
+
+enum class ParseState {
+  // Parse result is out. It could be HTTP CONNECT or not.
+  Done,
+  // Parser expects more data.
+  Continue,
+  // Parser reports unrecoverable error.
+  Error
+};
+/**
+ * Global configuration for CONNECT handler.
+ */
+class Config {
+public:
+  Config(Stats::Scope& scope);
+
+  const ConnectHandlerStats& stats() const { return stats_; }
+
+  static constexpr uint32_t MAX_INSPECT_SIZE = 8192;
```

Comment:
What's the point of having it as a part of public interface? Could it be just a static storage variable?

Reply:
I followed the patther of http and tls inspectors, are you suggesting setting MAX_INSPECT_SIZE as a static variable in connect_handler.h file?

Reply:
It could be a constexpr in the anonymous namespace of connect_handler.cc. `static` is redundant in this case.

But now I see it's used in the declaration of `Filter`... Let's have it unchanged then.